### PR TITLE
[Merged by Bors] - chore: add some simp and fun_prop attributes

### PR DESCRIPTION
**Pull request by fpvandoorn**
* Also add a test file that uses the `simp` lemmas.
* Add/move the `fun_prop` lemmas to the compositional lemmas.

[Zulip](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Derivative.20cannot.20simp)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: MathlibTest/Deriv.lean
Comment (grunweg) [inline MathlibTest/Deriv.lean:20]: Can you use the new +option syntax here? Also below?

File: MathlibTest/Deriv.lean
Comment (grunweg) [inline MathlibTest/Deriv.lean:7]: `fun x`, please

**Review by grunweg — APPROVED** [review]
Looks innocuous, and useful enough. Two nits; please address before merging. Thanks!
maintainer delegate

File: MathlibTest/Deriv.lean
Comment (fpvandoorn) [reply MathlibTest/Deriv.lean]: I had to dig to the actual PR description https://github.com/leanprover/lean4/pull/5883 to figure out that this had shorter syntax.

**Review by j-loreaux — COMMENTED** [review]
It turns out that these are not actually the most useful lemmas to tag with `fun_prop` in practice (although doing these doesn't really hurt; I know mainly what you are adding is `simp`. What you actually want are the *compositional* forms of all of these lemmas. That is: we actually want things like [Differentiable.exp](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/SpecialFunctions/ExpDeriv.html#Differentiable.exp) to be marked with `fun_prop`. While we're here do you mind marking those accordingly?

**Comment by fpvandoorn** [comment]
> It turns out that these are not actually the most useful lemmas to tag with `fun_prop` in practice (although doing these doesn't really hurt; I know mainly what you are adding is `simp`. What you actually want are the _compositional_ forms of all of these lemmas. That is: we actually want things like [Differentiable.exp](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/SpecialFunctions/ExpDeriv.html#Differentiable.exp) to be marked with `fun_prop`. While we're here do you mind marking those accordingly?

I've noticed that the number of failures of `fun_prop` actually increases when tracing its output, so I expect it is important not to tag them both for performance reasons.

I applied the `fun_prop` attributes to the compositional forms only, and therefore I removed some `fun_prop` attributes in `Trigonometric.Deriv`.

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**
